AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2256,41 +2319,48 @@ - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
   return [[ASLayoutSpec alloc] init];
 }
 
+- (void)setLayoutSpecBlock:(ASLayoutSpecBlock)layoutSpecBlock
+{
+  // For now there should never be a overwrite of layoutSpecThatFits: and a layoutSpecThatFitsBlock: be provided
+  ASDisplayNodeAssert(!(_methodOverrides & ASDisplayNodeMethodOverrideLayoutSpecThatFits), @"Overwriting layoutSpecThatFits: and providing a layoutSpecBlock block is currently not supported");
+
+  ASDN::MutexLocker l(__instanceLock__);
+  _layoutSpecBlock = [layoutSpecBlock copy];
+}
+
+- (ASLayoutSpecBlock)layoutSpecBlock
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _layoutSpecBlock;
+}
+
 - (ASLayout *)calculatedLayout
 {
   ASDN::MutexLocker l(__instanceLock__);
-  return _calculatedLayout;
+  return _calculatedDisplayNodeLayout->layout;
 }
 
-- (void)setCalculatedLayout:(ASLayout *)calculatedLayout
+- (void)setCalculatedDisplayNodeLayout:(std::shared_ptr<ASDisplayNodeLayout>)displayNodeLayout
```

Comment:
`ASDisplayNodeLayout` does not implement equality at the moment. If we would like to add that it would be way more work as we would have to add it to `ASLayout` as well as `ASLayoutable`